### PR TITLE
OLS-589: Specify host in all cases

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Specify host in all cases

## Type of change

- [ ] Refactor
- [ ] New feature
- [x] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #[OLS-589](https://issues.redhat.com//browse/OLS-589)
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: runner.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def configure_gradio_ui_envs() -> None:
     config.init_query_filter()
     config.init_vector_index()
 
-    host = "localhost" if config.dev_config.run_on_localhost else None
+    host = "localhost" if config.dev_config.run_on_localhost else "0.0.0.0"  # noqa S104
```
Comment (onmete) [inline runner.py:61]: `None` was previously transformed to `127.0.0.1`. Is this change intended?

Comment (tisnik) [reply runner.py:61]: was where?

Comment (onmete) [reply runner.py:61]: in `uvicorn.run` when the `host=None`

Comment (tisnik) [reply runner.py:61]: when it is *not specified*, it is 127.0.0.1, not when we pass *None* in string param
